src/XMonad/Operations.hs

```diff
@@ -431,17 +431,24 @@ sendMessage a = windowBracket_ $ do
 
 -- | Send a message to all layouts, without refreshing.
 broadcastMessage :: Message a => a -> X ()
-broadcastMessage a = withWindowSet $ \ws -> do
-   let c = W.workspace . W.current $ ws
-       v = map W.workspace . W.visible $ ws
-       h = W.hidden ws
-   mapM_ (sendMessageWithNoRefresh a) (c : v ++ h)
+broadcastMessage = filterMessageWithNoRefresh (const True)
 
 -- | Send a message to a layout, without refreshing.
 sendMessageWithNoRefresh :: Message a => a -> W.Workspace WorkspaceId (Layout Window) Window -> X ()
 sendMessageWithNoRefresh a w =
-    handleMessage (W.layout w) (SomeMessage a) `catchX` return Nothing >>=
-    updateLayout  (W.tag w)
+    filterMessageWithNoRefresh (on (==) W.tag w) a
```

Comment:
I don't know if this is just me, but I always find `on` to be terribly confusing if it's not written in an infix style, because when it reads like "do something on `(==)`" instead of "do `(==)` on something`

Reply:
OK, switched to infix `on`. I'm used to thinking of infix vs. prefix as a way to save on parentheses (in this case there are fewer prefix). But I see how, as a word, using it the 'wrong' way seems nongrammatical.